### DISCUSSION TITLE
Is the calculation able to handle recipes that produce multiple kinds of items?

### DISCUSSION THREAD

**Discussion by Epicmasonic** [AsherGlick/ResourceCalculator#113] [Q&A] [answered]
I made a fork to add a calculator for a Minecraft modpack and ran into a recipe that makes both Cobblestone and Nether Quartz Dust. Can the calculator use recipes like this?

**Comment by AsherGlick** [comment, marked as answer]
The current solver can only truly output a single type of item. I have some designs for how to upgrade the solver to be able to handle much more complex directed cyclic graphs instead of just directed trees but don't expect that any time soon. Subnautica also has this issue where some recipes for diving suits produce multiple outputs.

**Comment by Schop0** [comment]
Outworld Station also has recipes with multiple outputs. I understand that an upgrade to the solver can be quite complex. Possibly not worth the effort at all anytime soon.

A very useful intermediate step would be to just visualise additional outputs and ignore them. This allows the user to see them and account for it manually.

I'd love to hear if you are interested in implementing this. If not, I might have a go at it.

**Reply by AsherGlick** [reply]
Can you make an ms paint like mockup of what you think it would look like?

**Reply by Schop0** [reply]
Something like this maybe.
![image](https://github.com/user-attachments/assets/e3c80caa-46f0-4c62-b073-2aeef37e2206)
